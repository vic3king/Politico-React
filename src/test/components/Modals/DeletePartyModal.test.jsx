import React from 'react';
import { shallow } from 'enzyme';
import DeletePartModal from '../../../components/Modals/DeletePartyModal';

const hide = jest.fn();
const handleDelete = jest.fn();
let wrapper;
describe('InterestFormModal component', () => {
  global.fetch = jest.fn();
  global.localStorage.setItem('user', '{}');

  beforeEach(() => {
    wrapper = shallow(
      <DeletePartModal handleDelete={handleDelete} hide={hide} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('hide method', () => {
    wrapper = shallow(
      <DeletePartModal handleDelete={handleDelete} hide={hide} />
    );
    const instance = wrapper.instance();
    let event;
    it('should call preventDefault on event', async () => {
      event = {
        preventDefault: jest.fn(),
      };

      await instance.hide(event);
      expect(event.preventDefault).toBeCalled();
    });
  });

  describe('handleDelete method', () => {
    wrapper = shallow(
      <DeletePartModal handleDelete={handleDelete} hide={hide} />
    );
    const instance = wrapper.instance();
    let event;
    it('should call preventDefault on event', async () => {
      event = {
        preventDefault: jest.fn(),
      };

      await instance.handleDelete(event);
      expect(event.preventDefault).toBeCalled();
    });
  });
});
