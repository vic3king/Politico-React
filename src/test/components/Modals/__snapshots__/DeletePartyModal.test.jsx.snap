// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterestFormModal component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeletePartModal
    handleDelete={[MockFunction]}
    hide={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="modalDelete"
      >
        <p>
          Are you sure you want to delete?
        </p>
        <button
          className="modalyes"
          onClick={[Function]}
          type="submit"
        >
          Yes
        </button>
        <button
          className="modalno"
          type="submit"
        >
          No
        </button>
        <button
          className="modal-close"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Are you sure you want to delete?
          </p>,
          <button
            className="modalyes"
            onClick={[Function]}
            type="submit"
          >
            Yes
          </button>,
          <button
            className="modalno"
            type="submit"
          >
            No
          </button>,
          <button
            className="modal-close"
            onClick={[Function]}
            type="button"
          >
            X
          </button>,
        ],
        "className": "modalDelete",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Are you sure you want to delete?",
          },
          "ref": null,
          "rendered": "Are you sure you want to delete?",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Yes",
            "className": "modalyes",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "Yes",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No",
            "className": "modalno",
            "type": "submit",
          },
          "ref": null,
          "rendered": "No",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "modal-close",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "X",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="modalDelete"
        >
          <p>
            Are you sure you want to delete?
          </p>
          <button
            className="modalyes"
            onClick={[Function]}
            type="submit"
          >
            Yes
          </button>
          <button
            className="modalno"
            type="submit"
          >
            No
          </button>
          <button
            className="modal-close"
            onClick={[Function]}
            type="button"
          >
            X
          </button>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Are you sure you want to delete?
            </p>,
            <button
              className="modalyes"
              onClick={[Function]}
              type="submit"
            >
              Yes
            </button>,
            <button
              className="modalno"
              type="submit"
            >
              No
            </button>,
            <button
              className="modal-close"
              onClick={[Function]}
              type="button"
            >
              X
            </button>,
          ],
          "className": "modalDelete",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Are you sure you want to delete?",
            },
            "ref": null,
            "rendered": "Are you sure you want to delete?",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Yes",
              "className": "modalyes",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Yes",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No",
              "className": "modalno",
              "type": "submit",
            },
            "ref": null,
            "rendered": "No",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "X",
              "className": "modal-close",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "X",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
